lib/signaling/v2/iceconnectionmonitor.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const { ICE_ACTIVITY_CHECK_PERIOD_MS, ICE_INACTIVITY_THRESHOLD_MS } = require('../../util/constants');
+
+/**
+ * Monitors a {@link RTCPeerConnection}'s stats and notifies
+ * caller when inactivity is detected.
+ */
+class IceConnectionMonitor {
+  /**
+   * Construct an {@link IceConnectionMonitor}.
+   * @param {RTCPeerConnection} peerConnection
+   * @param {object} [options]
+   */
+  constructor(peerConnection, options) {
+    options = Object.assign({
+      activityCheckPeriodMs: ICE_ACTIVITY_CHECK_PERIOD_MS,
+      inactivityThresholdMs: ICE_INACTIVITY_THRESHOLD_MS,
+    }, options);
+
+    Object.defineProperties(this, {
+      _activityCheckPeriodMs: {
+        value: options.activityCheckPeriodMs
+      },
+      _inactivityThresholdMs: {
+        value: options.inactivityThresholdMs
+      },
+      _lastActiveStats: {
+        value: null,
+        writable: true
+      },
+      _peerConnection: {
+        value: peerConnection
+      },
+      _timer: {
+        value: null,
+        writable: true,
+      }
+    });
+  }
+
+  /**
+   * Get ICE connection stats, and extract received and send bytes.
+   * @returns Promise<?{timestamp, bytesReceived, bytesSent}>
+   */
+  _getIceConnectionStats() {
+    return this._peerConnection.getStats().then(stats => Array.from(stats.values()).find(stat => {
+      return stat.type === 'candidate-pair' && stat.nominated;
+    })).catch(() => {
+      return null;
+    });
+  }
+
+  /**
+   * Start monitoring the ICE connection.
+   * Monitors bytes received on active ice connection pair,
+   * invokes onIceConnectionInactive when inactivity is detected.
+   * @param {function} onIceConnectionInactive
+   */
+  start(onIceConnectionInactive) {
+    this.stop();
+    this._timer = setInterval(() => {
+      this._getIceConnectionStats().then(iceStats => {
+        if (!iceStats) {
+          return;
+        }
+
+        // if change detected in bytesReceived mark it as lastActiveiy.
```

Comment:
@makarandp0 I think the code is self-explanatory, so we don't need the comment here.